### PR TITLE
Route `ServiceLevelBurnRateTooHigh` alert using `application.giantswarm.io/team` label

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/22290

This PR aims at using dynamic routing for `ServiceLevelBurnRateTooHigh`. (see [article](https://tech.loveholidays.com/dynamic-alert-routing-with-prometheus-and-alertmanager-f6a919edb5f8))

The idea basically to be able to route alert based on labels given to k8s resources. To achieve this we need:

1. Label our app with team name ([PR](https://github.com/giantswarm/prometheus-operator-app/pull/191/files#diff-e27896bc0794dd5a2ff25c437b60bb597c4b52f02e224d83e61d1bacb7d1f6f3R12))
2. Have this label in prometheus (thanks to kube-state-metrics, see [PR](https://github.com/giantswarm/kube-state-metrics-app/pull/138))
3. Add this label into our alert (this PR)
4. Route to different team based on tag/label (already done in [OpsGenie](https://giantswarm.app.opsgenie.com/teams/dashboard/bbbc9d91-93fa-4882-8b07-84bebea41508/main))

There are 3 main changes in this PR:

1. I added the `application.giantswarm.io/team` label to every managed-app slo rules. This was achieved by wrapping the current expression with
```
kube_daemonset_labels * on (daemonset, namespace) group_right(label_application_giantswarm_io_team) (
  current_expression
)
```
2. Adding the `label_application_giantswarm_io_team` to every `slo_errors*` recording rules
3. Adding the `label_application_giantswarm_io_team` as `team` label into `ServiceLevelBurnRateTooHigh` alert

### CODE REVIEW

**Comment by TheoBrigitte** [comment]
Tested and working

image

https://giantswarm.app.opsgenie.com/alert/detail/2c1ff9c8-77f2-4c66-802b-92a9acf3fee2-1654695798458/details

File: helm/prometheus-rules/templates/alerting-rules/service-level.rules.yml [hunk 1]
```diff
@@ -34,3 +34,4 @@ spec:
         cancel_if_cluster_has_no_workers: "true"
         cancel_if_outside_working_hours: {{ include "workingHoursOnly" . }}
         severity: page
+        team: '{{`{{ $labels.label_giantswarm_io_monitoring_team }}`}}'
```
Comment (whites11) [inline helm/prometheus-rules/templates/alerting-rules/service-level.rules.yml:37]: neat trick!

**Review by whites11 — APPROVED** [review]
GREAT STUFF

**Comment by TheoBrigitte** [comment]
@giantswarm/team-atlas WDYT ?

**Comment by hervenicol** [comment]
:+1: :clap:  for the logic.
However, I know some Prometheus are short on resources. What's the amount of new timeseries we add with these new metrics? What's the impact on cardinality?

**Comment by TheoBrigitte** [comment]
> +1 clap for the logic. However, I know some Prometheus are short on resources. What's the amount of new timeseries we add with these new metrics? What's the impact on cardinality?

Cardinality does not change we just have a new label showing up when its present.

**Comment by QuentinBisson** [comment]
As discussed in the prometheus-operator-app PR, I would use a team label, and not a monitoring_team label but we can merge this now and rewrite the label later to fix the current issue :)

**Review by QuentinBisson — APPROVED** [review]

**Comment by TheoBrigitte** [comment]
> > +1 clap for the logic. However, I know some Prometheus are short on resources. What's the amount of new timeseries we add with these new metrics? What's the impact on cardinality?
> 
> Cardinality does not change we just have a new label showing up when its present.

My knowledge about cardinality was wrong. So cardinality will change, adding the team label on the SLO queries should not raise cardinality too much. But having this label on every deployment, statefulset, and daemonset (in https://github.com/giantswarm/kube-state-metrics-app/pull/138) might be a bigger problem. Will investigate.

> As discussed in the prometheus-operator-app PR, I would use a team label, and not a monitoring_team label but we can merge this now and rewrite the label later to fix the current issue :) As Chris said, I agree this is great stuff

Thanks, I mainly wanted to get feedback on the logic itself. I will investigate which label we should use now.

**Comment by TheoBrigitte** [comment]
Tested and working with the new label

image

**Comment by QuentinBisson** [comment]
I was thinking of relabelling application.giantswarm.io/team into team if that's possible :)

**Comment by TheoBrigitte** [comment]
We agreed to do relabelinbg later if needed

**Pull request merged**